cmd/kube-proxy/app/server.go

```diff
@@ -814,3 +818,36 @@ func (s *ProxyServer) CleanupAndExit() error {
 
 	return nil
 }
+
+// detectNodeIP returns the nodeIP used by the proxier
+// The order of precedence is:
+// 1. config.bindAddress if bindAddress is not 0.0.0.0 or ::
+// 2. the primary IP from the Node object, if set
+// 3. if no IP is found it defaults to 127.0.0.1 and IPv4
+func detectNodeIP(client clientset.Interface, hostname, bindAddress string) net.IP {
```

Comment:
There is a util pkg under proxy where there is a function that returns all addresses for a node as `sets.String{}` -- Why are we using a different approach here?

Reply:
we don't need all addresses of a node, we need to have a consistent way to detect the primary ip of the node to deal with the kube-proxy nodeIP chaos, related to https://github.com/kubernetes/kubernetes/pull/104691/files#r700086018
and explained https://github.com/kubernetes/kubernetes/issues/95515